core/src/main/java/org/apache/carbondata/core/view/MVProvider.java

```diff
@@ -540,7 +540,7 @@ private void checkAndReloadSchemas(MVManager viewManager, boolean touchFile)
       }
     }
 
-    private void touchMDTFile() throws IOException {
+    private synchronized void touchMDTFile() throws IOException {
```

Comment:
In saveSchema() method of same file, 
checkAndReloadSchemas(viewManager, true);
        touchMDTFile();

line 432: after passing touch = true, again we are calling touchMDTFile, is itrequired?

Reply:
@Indhumathi27 , @vikramahuja1001

Reply:
@ajantha-bhat yes, it is required. On second MV creation, have to update last modified time of schema INdex file

Reply:
checkAndReloadSchemas, better to return a flag whether it has touched MDT or not, if false then only we can call touchMDT

Reply:
done